skyfield/sgp4lib.py

```diff
@@ -153,6 +153,30 @@ def ITRF_position_velocity_error(self, t):
         rITRF, vITRF = TEME_to_ITRF(t.ut1, rTEME, vTEME)
         return rITRF, vITRF, error
 
+    def is_sunlit(self, ephemeris, times):
+        """Calculates if the satellite is currently sunlit (True/False) at each time in times. Returns an array of booleans."""
+        Re = 6378.137  # Equatorial radius of the Earth in km
+        earth = ephemeris['earth']
+        sun = ephemeris['sun']
+        sat = earth + self
+
+        sunpos, earthpos, satpos = [
+            thing.at(times).position.km for thing in (sun, earth, sat)
+        ]
+        sunearth, sunsat = earthpos-sunpos, satpos-sunpos
+        sunearthnorm, sunsatnorm = [
```

Comment:
Let's also unwrap this loop, for simplicity in both reading and when Python executes it. And I think you can use `skyfield.fuctions.length_of()` here instead of the raw formula — as well as on the following lines. It will make the code easier for me, at least, to read and maintain. Thanks!